### ISSUE TITLE
Release binary issue - "This SDK is not supported by the compiler"

### ISSUE THREAD

**Issue by MarkWatson16** [amazon-connect/amazon-connect-chat-ios#43]
Hi @mrajatttt and @mliao95, looks like I'm still facing the issue. I'll actually say what the specific error is this time 😅 

So we’re having an issue using the pre-compiled framework that’s put on the release pages for the repo. It seems to be locked to specific compiler version so unless you use that exact version too, it won’t compile. Even with a more recent compiler.

I just tried adding the 1.0.2 xcframework from the release pages (1.0.2-public.zip) to a fresh sample project but I get this error.

> Failed to build module 'AmazonConnectChatIOS'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)', while this compiler is 'Apple Swift version 6.0 effective-5.10 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)'). Please select a toolchain which matches the SDK.

![Screenshot 2024-11-22 at 11 13 30](https://github.com/user-attachments/assets/a7809879-4a25-44d0-8a17-e90bb008fa13)

I also tried to add it to our project and got the same issue. I had googled around was hoping that distribution setting would have fixed it but sadly hasn't.

I’m not sure of other solutions to fix it at this point, some suggest a potential issue with cocoapods but we’re not using that on our side. We’re using Xcode 16 as of last week but I had the issue with 15.2 as well.

I think it should possible to replicate it by creating the framework using a specific tool version, and then just trying to use that on a fresh sample project and with a different Xcode or tool version. Anything using a different tool set.

**Comment by MarkWatson16** [comment]
I also get this error with the PrivacyInfo when I don't import the framework into a file. Oddly I can use SPM to build it myself, it's just the release binary that seems to have this issue.

> unable to read property list from privacy file: /Users/myusername/Library/Developer/Xcode/DerivedData/MyApp-gqgalfwffuowprdilppdimcgpgyv/Build/Products/Debug-iphonesimulator/MyApp.app/Frameworks/AmazonConnectChatIOS.framework/PrivacyInfo.xcprivacy: The operation couldn’t be completed. (XCBUtil.PropertyListConversionError error 2.)

**Comment by mrajatttt** [comment]
Hi @MarkWatson16, requesting you to verify above in latest version. https://github.com/amazon-connect/amazon-connect-chat-ios/releases/tag/v1.0.3

Also updated this for users who use binaries: https://github.com/amazon-connect/amazon-connect-chat-ios?tab=readme-ov-file#download-binaries-directly-from-github-releases

**Comment by MarkWatson16** [comment]
Hey @mrajatttt, the privacy issue is fixed, thanks. I'm not getting that error anymore.

I still have the toolchain error when using the binary but I don't see a fix for that in your changes

**Comment by mrajatttt** [comment]
Hi @MarkWatson16, What xcode version you are using? Can you please tell me your xcode and swift version. 

Second, Did you make sure that you have imported other required packages? 

During my testing, I faced similar issues and when i imported other required frameworks (AWSConnectParticipant and AWSCore), the error went away.

**Comment by MarkWatson16** [comment]
Hi @mrajatttt, we're using Xcode 16.0, swift 5.10 on our project. I add the framework to a sample project with swift 6 though with the same issue

Ah I see, I think that would work if I grab the other frameworks from the carthage bundled .zip on the SDK page. I might be able to bundle them into a single package on our side too, I'll let you know by Monday

**Comment by mrajatttt** [comment]
Yes, thank you. While working on this issue, I understood that when we manually import binaries, dependencies are not bundled along like they would be with SPM or pods. This mismatch was the reason for the issue, at least in my case.

**Comment by MarkWatson16** [comment]
Hey sorry for the delay, I've tried the importing and that worked! I grabbed the xcframeworks for `AWSCore` and `AWSConnectParticipant` from the Carthage framework zip on the releases [here](https://github.com/aws-amplify/aws-sdk-ios/releases), added them and it built successfully.

There is an issue though that grabbing the xcframeworks from that carthage framework means they aren't signed. I assume you have to build through Carthage to get the signed versions.
|Signed - AmazonConnectChat|Unsigned - AWSCore|
|---|---|
|<img width="250" alt="4" src="https://github.com/user-attachments/assets/d57781df-6982-45ed-9904-e6b079d76528">|<img width="250" alt="4" src="https://github.com/user-attachments/assets/a2c7de0a-2877-4af8-941c-5e1fd0b82e93">|

I've tried reaching out to our platform team, who manage the dependencies, but they must be busy or something as I'm not getting a reply which was why I had a delay. I wanted to check if we can use the unsigned ones too but my guess was they could fail on the release build or app store review/upload. It does mention they have to be signed in our internal instructions for new dependencies.

So I think we just have go with the SPM build route because getting them signed might mean you have to separate them from the main AWS SDK repo or changing the build process to build independent xcframeworks for everything in there each release

**Comment by mrajatttt** [comment]
Hi @MarkWatson16, as far as I understand, having unsigned libraries could cause your application to fail during release.

Regarding signing AWSCore and AWSConnectParticipant, we don’t own these SDKs, and they’re part of https://github.com/aws-amplify/aws-sdk-ios/tree/main, which is already signed. Therefore, it’s not possible for us to sign those two on our own.

I believe the SPM route is the most suitable option because it effectively manages dependencies on its own.

**Comment by MarkWatson16** [comment]
Yep @mrajatttt, I agree. We've gone with the SPM route, I'll let you know if anything else comes up. Thanks for the help!
